mathics/doc/latex_doc.py

```diff
@@ -666,258 +655,22 @@ def latex(self, doc_data: dict):
         )
 
 
-class LaTeXMathicsMainDocumentation(MathicsMainDocumentation):
+class LaTeXMathicsDocumentation(Documentation):
     def __init__(self, want_sorting=False):
+        self.doc_chapter_fn = LaTeXDocChapter
         self.doc_dir = settings.DOC_DIR
+        self.doc_fn = LaTeXDoc
+        self.doc_data_file = settings.get_doc_latex_data_path(should_be_readable=True)
+        self.doc_guide_section_fn = LaTeXDocGuideSection
+        self.doc_part_fn = LaTeXDocPart
+        self.doc_section_fn = LaTeXDocSection
+        self.doc_subsection_fn = LaTeXDocSubsection
         self.latex_pcl_path = settings.DOC_LATEX_DATA_PCL
         self.parts = []
         self.parts_by_slug = {}
-        self.pymathics_doc_loaded = False
-        self.doc_data_file = settings.get_doc_latex_data_path(should_be_readable=True)
         self.title = "Overview"
-        files = listdir(self.doc_dir)
```

Comment:
A programmer who write such a long piece of code like this should be graduated out of programming.